### DISCUSSION TITLE
How do I properly add Chicago to `SERVER_CITIES` for AirVPN?

### DISCUSSION THREAD

**Discussion by raddevon** [qdm12/gluetun#2501] [Q&A]
I'm configuring my container for AirVPN and want to provide a list of cities via the `SERVER_CITIES` env var. I want to add Chicago, Illinois to my list, but I'm not sure how since the value itself contains a comma and the list should be comma-separated.

How should I do this? I've tried a number of approaches like using just `Chicago` (which led me to believe it wants `Chicago, Illinois` based on the error message), `Chicago\, Illinois`, `"Chicago, Illinois"`, and `Chicago Illinois`, all of which are failing. Oddly, in [servers.json](https://raw.githubusercontent.com/qdm12/gluetun/refs/heads/master/internal/storage/servers.json), it's listed as `Chicago Illinois` even though that doesn't seem to work and the error suggests it wants `Chicago, Illinois`. Not sure what else to try.

Permutation:
```
      - SERVER_CITIES=New York City,Pennsylvania,Chicago
```

Result:

> 2024-09-30T19:40:06Z ERROR VPN settings: provider settings: server selection: the city specified is not valid: value is not one of the possible choices: value "chicago", choices available are Alblasserdam, Atlanta, Georgia, Auckland, Belgrade, Berlin, Bern, Brussels, Bucharest, Chicago, Illinois, Dallas, Texas, Frankfurt, Fremont, California, Jacksonville, Florida, Kiev, London, Los Angeles, Madrid, Manchester, Miami, Montreal, Munich, New York City, Oslo, Pennsylvania, Phoenix, Arizona, Prague, Riga, Sao Paulo, Singapore, Sofia, Stockholm, Tallinn, Tokyo, Toronto, Ontario, Uppsala, Vancouver, Vienna, Zurich

---

Permutation:
```
      - SERVER_CITIES=New York City,Pennsylvania,"Chicago, Illinois"
```

Result:

> 2024-09-30T19:40:06Z ERROR VPN settings: provider settings: server selection: the city specified is not valid: value is not one of the possible choices: value "\\"chicago", choices available are Alblasserdam, Atlanta, Georgia, Auckland, Belgrade, Berlin, Bern, Brussels, Bucharest, Chicago, Illinois, Dallas, Texas, Frankfurt, Fremont, California, Jacksonville, Florida, Kiev, London, Los Angeles, Madrid, Manchester, Miami, Montreal, Munich, New York City, Oslo, Pennsylvania, Phoenix, Arizona, Prague, Riga, Sao Paulo, Singapore, Sofia, Stockholm, Tallinn, Tokyo, Toronto, Ontario, Uppsala, Vancouver, Vienna, Zurich

---

Permutation:
```
      - SERVER_CITIES=New York City,Pennsylvania,Chicago\, Illinois
```

Result:

> 2024-09-30T19:40:06Z ERROR VPN settings: provider settings: server selection: the city specified is not valid: value is not one of the possible choices: value "chicago\\\\", choices available are Alblasserdam, Atlanta, Georgia, Auckland, Belgrade, Berlin, Bern, Brussels, Bucharest, Chicago, Illinois, Dallas, Texas, Frankfurt, Fremont, California, Jacksonville, Florida, Kiev, London, Los Angeles, Madrid, Manchester, Miami, Montreal, Munich, New York City, Oslo, Pennsylvania, Phoenix, Arizona, Prague, Riga, Sao Paulo, Singapore, Sofia, Stockholm, Tallinn, Tokyo, Toronto, Ontario, Uppsala, Vancouver, Vienna, Zurich

---

Permutation:
```
      - SERVER_CITIES=New York City,Pennsylvania,Chicago Illinois
```

Result:

> 2024-09-30T19:40:06Z ERROR VPN settings: provider settings: server selection: the city specified is not valid: value is not one of the possible choices: value "chicago illinois", choices available are Alblasserdam, Atlanta, Georgia, Auckland, Belgrade, Berlin, Bern, Brussels, Bucharest, Chicago, Illinois, Dallas, Texas, Frankfurt, Fremont, California, Jacksonville, Florida, Kiev, London, Los Angeles, Madrid, Manchester, Miami, Montreal, Munich, New York City, Oslo, Pennsylvania, Phoenix, Arizona, Prague, Riga, Sao Paulo, Singapore, Sofia, Stockholm, Tallinn, Tokyo, Toronto, Ontario, Uppsala, Vancouver, Vienna, Zurich

**Comment by starrbuck68** [comment]
Have you tried:

`SERVER_CITIES='New York City,Pennsylvania,Chicago Illinois'`

Pennsylvania is not a city so I'm confused as to why it's in their list.

There are tons of servers in New York City and Chicago, so if you wanted to use one this would surely work:

`SERVER_CITIES='New York City'`

or

`SERVER_CITIES='Chicago Illinois'`

**Reply by raddevon** [reply]
I believe I tried double quotes around everything but not single. What's odd is that I've never had to use quotes around New York City, and it has always worked. 🤔

I was also confused how/why Pennsylvania made it to the cities list. 😅

**Reply by raddevon** [reply]
For now, I have it working with just NYC and Pennsylvania, so I don't think I'm going to touch it. Those servers are going to be further away from me anyway, so I don't think I need them.

**Reply by sofakng** [reply]
I'm having the same problem I think.  I can't get SERVER_CITIES to work at all it seems and it just ignores the value.

**Reply by raddevon** [reply]
@sofakng Sorry for the late reply. I think your problem may be different than mine. `SERVER_CITIES` works for me if I just take out Chicago. My only problem seems to be that I don't know how the app expects me to specify Chicago.
